src/core/device.c

```diff
@@ -1075,6 +1079,98 @@ bool device_shall_be_bound_by(Unit *device, Unit *u) {
         return DEVICE(device)->bind_mounts;
 }
 
+/**
+ * While booting up it is likely that systemd-udevd is started after a couple of uevents were already
+ * triggered by the kernel and are missed for that reason. To fix that up device_start() is used to
+ * re-trigger the corresponding uevent of the particular device unit.
+ *
+ * The re-trigger is only done in case the device is not following another device, was never in plugged
+ * state before, is in the dead state, but is actually plugged (corresponding sysfs entries are existent).
+ *
+ * Note: This accounts only for device units which are explicitly referenced as dependency by other units
+ *       and are tagged in udev rules (TAG+="systemd").
+ *
+ *       This might only be useful in highly embedded environments, e.g. automotive control units, where
+ *       existent devices and the whole dependency tree are known, the ecosystem is well-defined and startup
+ *       performance matters.
+ *
+ *       For desktop and server environments systemd-udev-trigger.service should still be used.
+ */
+static int device_start(Unit *u) {
+        Device *d = DEVICE(u);
+        _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
+        _cleanup_free_ char *path = NULL;
+        int r;
+
+        assert(d);
+
+        if (device_following(u))
+                return 0;
+
+        if (!(d->state == DEVICE_DEAD))
+                return 0;
+
+        if (d->plugged_once)
+                return 0;
+
+        r = unit_name_to_path(u->id, &path);
+        if (r)
```

Comment:
It returns `0` on success...
By the way, the return values are not used in most places, so,
```
if (unit_name_to_path(u->id, &path) < 0)
        return log_unit_warning_errno(...);
```
should be ok.

Reply:
Since in any error case always -EBADR is returned I thought it might be useful to get the actual errno of the function which failed printed.

Do you want me to remove that for any `return log_unit_*_errno()` statements?